### DISCUSSION TITLE
Store data in a structure of folders and subfolders, so that data are 'naturally' sorted

### DISCUSSION THREAD

**Discussion by GitHunter0** [xiaodaigh/disk.frame#335] [Ideas]
Hi folks, this is a follow up of the discussion I had here with @xiaodaigh : https://github.com/xiaodaigh/disk.frame/issues/327#issuecomment-785829324

This would allow to work with longitudinal/panel data without the need to apply the very expensive 'hard sorting' operation.

The picture below shows a tree example:
![image](https://user-images.githubusercontent.com/26294221/115489350-75010e00-a232-11eb-9bc8-89829f68b6e9.png)
That way the id variable (country) and date variable (year) would be already sorted from the start.

That is precisely the strategy adopted by Apache Arrow, as it is demonstrated here https://cran.r-project.org/web/packages/arrow/vignettes/dataset.html

This feature in my opinion is one of the last pieces which would allow a more frequent/universal usage of disk.frame (which is already a phenomenal package)

**Comment by xiaodaigh** [comment]
this is a good idea! I wonder if I should prioritise this above the NSE rewrite. Maybe I should.

I am currently working on fixing JDF.jl as my OS commitment, after that I can come back to disk.frame.

**Reply by GitHunter0** [reply]
Hey @xiaodaigh , no doubt NSE is also a very relevant issue. However, for the most frequent use cases I believe this issue is more important indeed.

Nice to know, good luck with your Julia projects too
